corona-warn-app/cwa-app-ios
# [Feature] Generate new database key after resets and reinstalls

## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description
After a manual reset or after an app reinstall the database key should be changed. This PR handles the check if it is a first app start (first installation, reinstallation) and the rekeying after a reset with storing the new db key in the keychain. 

## How to test
Start the App and everything should work as exspected. Afterwards you can manually reset the app and see if everything is working as exspected
